src/main/java/teammates/ui/controller/FeedbackSubmissionEditSaveAction.java

```diff
@@ -297,63 +295,69 @@ private FeedbackResponseAttributes extractFeedbackResponseData(
                 Const.ParamsNames.FEEDBACK_QUESTION_ID + "-" + questionIndx);
         Assumption.assertPostParamNotNull(Const.ParamsNames.FEEDBACK_QUESTION_ID + "-" + questionIndx,
-                response.feedbackQuestionId);
+                feedbackQuestionId);
         Assumption.assertEquals("feedbackQuestionId Mismatch", feedbackQuestionAttributes.getId(),
-                                response.feedbackQuestionId);
+                                feedbackQuestionId);
 
-        response.recipient = getRequestParamValue(
+        String recipient = getRequestParamValue(
                 Const.ParamsNames.FEEDBACK_RESPONSE_RECIPIENT + "-" + questionIndx + "-" + responseIndx);
         Assumption.assertPostParamNotNull(Const.ParamsNames.FEEDBACK_RESPONSE_RECIPIENT + "-" + questionIndx + "-"
-                + responseIndx, response.recipient);
+                + responseIndx, recipient);
 
         String feedbackQuestionType = getRequestParamValue(
                 Const.ParamsNames.FEEDBACK_QUESTION_TYPE + "-" + questionIndx);
         Assumption.assertPostParamNotNull(Const.ParamsNames.FEEDBACK_QUESTION_TYPE + "-" + questionIndx,
                 feedbackQuestionType);
-        response.feedbackQuestionType = FeedbackQuestionType.valueOf(feedbackQuestionType);
 
         FeedbackParticipantType recipientType = feedbackQuestionAttributes.recipientType;
+        String recipientSection;
         if (recipientType == FeedbackParticipantType.INSTRUCTORS || recipientType == FeedbackParticipantType.NONE) {
-            response.recipientSection = Const.DEFAULT_SECTION;
+            recipientSection = Const.DEFAULT_SECTION;
         } else if (recipientType == FeedbackParticipantType.TEAMS) {
-            response.recipientSection = logic.getSectionForTeam(courseId, response.recipient);
+            recipientSection = logic.getSectionForTeam(courseId, recipient);
         } else if (recipientType == FeedbackParticipantType.STUDENTS) {
-            StudentAttributes student = logic.getStudentForEmail(courseId, response.recipient);
-            response.recipientSection = student == null ? Const.DEFAULT_SECTION : student.section;
+            StudentAttributes student = logic.getStudentForEmail(courseId, recipient);
+            recipientSection = student == null ? Const.DEFAULT_SECTION : student.section;
         } else {
-            response.recipientSection = getUserSectionForCourse();
+            recipientSection = getUserSectionForCourse();
         }
 
         // This field can be null if the question is skipped
         String paramName = Const.ParamsNames.FEEDBACK_RESPONSE_TEXT + "-" + questionIndx + "-" + responseIndx;
         String[] answer = getRequestParamValues(paramName);
 
-        if (questionDetails.isQuestionSkipped(answer)) {
-            response.responseMetaData = new Text("");
-        } else {
-            FeedbackResponseDetails responseDetails =
+        FeedbackResponseDetails responseDetails = null;
```

Comment:
No, `responseDetails` will still be `Text("")`. It is set in `setResponseDetails()`.

Reply:
What I meant it still will be `null` here. And if we pass in a `null` as a function argument, how do we know the `null` is an empty value or that it is a `null` because of some error in computation?

In `setResponseDetails(..)` , the meaning of `new Text("")` and `null` got conflated however. Instead of handling a `null` error, it decides to hide the error and simply convert the `null` to `new Text("")`.

At a higher level, there are logs at a higher level to show such warnings `log.warning("Failed to extract response details.\n" + e.toString());`. Similarly, let's add a warning log for `setResponseDetails(..)`.

Reply:
The way the classes are structured makes it quite hard to do this in a clean way. In the end I had 2 `return` statement each differs from the other just by the way `responseDetails` is set, either by `withResponseMetaDataFromFeedbackResponseDetails` or by `withResponseMetaData`. I tried constructing `metaData` (which is a `Text`) from `FeedbackResponseDetails`, but it's not allowed without the `FeedbackResponseAttributes` itself which is `private`.

Reply:
Yeah, let's just go with this until we can address the other related issue of failing to extract response details.